### PR TITLE
Large navigation header

### PR DESCRIPTION
**Pull request by neb42**
* Add navigation header for large screens
* Link theme
* Fix deep linking

TODO:
* Tests

Future work:
* Implement logout flow in backend
* Small screen navigation header

### CODE REVIEW

File: frontend/packages/design-system/src/themes/Link.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  },
+  baseStyle: {
+    _text: {
+      fontSize: 'xs',
+      fontWeight: 'medium',
+      color: 'secondary.500',
+    },
+    _hover: {
+      _text: {
+        color: 'primary.500',
```
Comment (internetti) [inline frontend/packages/design-system/src/themes/Link.ts:13]: I assumed the active link would also be orange

File: frontend/packages/design-system/src/themes/Link.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+export default {
+  defaultProps: {
+    isUnderlined: false,
```
Comment (internetti) [inline frontend/packages/design-system/src/themes/Link.ts:3]: this doesn't seem to work

**Review by internetti — COMMENTED** [review]
in mobile the view, the navigator only shows one route, I'd expect at least also the logout button there

File: frontend/packages/design-system/src/themes/Link.ts (hunk 2 above)
Comment (neb42) [reply frontend/packages/design-system/src/themes/Link.ts:3]: This works for me. Did you build the design system?

File: frontend/packages/design-system/src/themes/Link.ts (hunk 1 above)
Comment (neb42) [reply frontend/packages/design-system/src/themes/Link.ts:13]: `_active` and `_focus` don't seem to do anything

File: frontend/apps/core-app/src/components/NavHeader/logout.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Platform } from 'react-native';
+import * as AuthSession from 'expo-auth-session';
+import { openAuthSessionAsync } from 'expo-web-browser';
+import Constants from 'expo-constants';
+
+const authorizationEndpoint = `${Constants.manifest?.extra?.issuer}/oauth2/sessions/logout`;
+const redirectUri = AuthSession.makeRedirectUri({ useProxy: false });
+
+export const logout = async () => {
```
[the remark this answers is not in the document]
Comment (neb42) [reply frontend/apps/core-app/src/components/NavHeader/logout.ts:9]: Yeah this is related to the comment in the pr description. I think the backend (login server) doesn't support logout yet. I've left this here for now and we can address the backend later

File: frontend/apps/core-app/src/components/NavHeader/LargeNavHeader.component.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import { logout } from './logout';
+
+const getHeaderTitle = (route: StackHeaderProps['route']): string => {
+  const routeName =
+    getFocusedRouteNameFromRoute(route) ?? routes.recipientsRoot.name;
+
+  return routes[routeName as keyof typeof routes].title;
+};
+
+export const LargeNavHeaderComponent: React.FC<StackHeaderProps> = ({
```
[the remark this answers is not in the document]
Comment (neb42) [reply frontend/apps/core-app/src/components/NavHeader/LargeNavHeader.component.tsx:18]: Yeah I went with Large/Small as it's based on screen size

File: frontend/apps/core-app/src/components/NavHeader/LargeNavHeader.component.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        />
+        <HStack>
+          <Link
+            to={{
+              screen: routes.recipientsRoot.name,
+              params: { screen: routes.recipientsList.name },
+            }}
+            mr={8}
+          >
+            Beneficiaries
```
[the remark this answers is not in the document]
Comment (neb42) [reply frontend/apps/core-app/src/components/NavHeader/LargeNavHeader.component.tsx:37]: Happy to add them to the route config, but not iterating over the config. Not all routes will be shown here so they should be added explicitly

**Comment by neb42** [comment]
> in mobile the view, the navigator only shows one route, I'd expect at least also the logout button there

Mobile will have a different header (drawer). I thought it's ok to leave it semi-broken for now as we mostly develop on web, but we should do another pr soon for the mobile header.

File: frontend/apps/core-app/src/components/NavHeader/logout.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (internetti) [reply frontend/apps/core-app/src/components/NavHeader/logout.ts:9]: could you create a ticket for that, so we don't forget?

File: frontend/apps/core-app/src/components/NavHeader/LargeNavHeader.component.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (internetti) [reply frontend/apps/core-app/src/components/NavHeader/LargeNavHeader.component.tsx:37]: okay, while we could add a flag to signal which ones are in the navigation, it's not that many routes, and I don't expect a lot of changes there, so fine by me

**Pull request merged**